pallets/randomness/src/vrf.rs

```diff
@@ -97,3 +99,14 @@ pub(crate) fn verify_and_set_output<T: Config>() {
 		RandomnessResults::<T>::insert(local_vrf_this_block, results);
 	}
 }
+
+pub(crate) fn set_previous_output<T: Config>(now: BlockNumberFor<T>) {
+	if let Some(randomness) = LocalVrfOutput::<T>::get() {
+		PreviousLocalVrfOutputs::<T>::mutate(|outputs| {
+			if outputs.len() > T::MaxPreviousLocalRandomness::get() as usize {
+				outputs.pop();
+			};
+			outputs.push((randomness.clone(), now));
+		});
+	}
+}
```

Comment:
if we do this in `on-initialize` after the output is set in storage, then there is no need for this DB read

Also it would be better weight accounting to do it in on-initialize, possibly just inside `vrf::set_output` and then re-run benchmarks.

Reply:
This is used to set the previous vrf, if done on `on_initialize` the storage would be set immediately (and hence not previous)

Reply:
It doesn't need to store the previous, it just needs to be stored.

Reply:
Yeah but given the following situation: Use previous blocks VRF as random seed.

If we set it in `on_initialize` and the `author-filter` requests a seed, then the storage will have a value (current VRF). Using `on_finalize` fixes this problem. Without that, think we'd need to put in some other parameters in place to offset the stored vrf.

Reply:
> Use previous blocks VRF as random seed.

So we just put it in storage and wait until it is previous by N blocks

> If we set it in on_initialize and the author-filter requests a seed, then the storage will have a value (current VRF). Using on_finalize fixes this problem. Without that, think we'd need to put in some other parameters in place to offset the stored vrf.

We don't use the current VRF for the current seed. We would use the old VRF that we put in storage N blocks ago,